components/kyma-metrics-collector/pkg/process/process.go

```diff
@@ -353,25 +353,59 @@ func isSuccess(status int) bool {
 	return false
 }
 
-func isClusterTrackable(runtime *kebruntime.RuntimeDTO) bool {
-	if runtime.Status.Provisioning != nil &&
-		runtime.Status.Provisioning.State == string(kebruntime.StateSucceeded) &&
-		runtime.Status.Deprovisioning == nil {
+// Helper function which returns true if the runtime state is trackable
+func isTrackableState(state kebruntime.State) bool {
+	switch state {
+	case kebruntime.StateSucceeded, kebruntime.StateError, kebruntime.StateUpgrading, kebruntime.StateUpdating:
 		return true
 	}
 	return false
 }
 
+func (p *Process) isClusterTrackable(runtime *kebruntime.RuntimeDTO) bool {
+	// Check if the cluster is in a trackable state
+	if isTrackableState(runtime.Status.State) ||
+		(runtime.Status.Provisioning != nil &&
+			runtime.Status.Provisioning.State == string(kebruntime.StateSucceeded) &&
+			runtime.Status.Deprovisioning == nil) {
+
+		p.namedLogger().With(log.KeySubAccountID, runtime.SubAccountID).With(log.KeyRuntimeID, runtime.RuntimeID).
+			Infof("Tracked cluster of state: %s, provisioning state: %s, no deprovisioning state.",
+				runtime.Status.State, runtime.Status.Provisioning.State)
+		return true
+	}
+
+	// Set provisioning and deprovisioning states if available
+	provisioning := getOrDefault(runtime.Status.Provisioning, "")
+	deprovisioning := getOrDefault(runtime.Status.Deprovisioning, "")
+
+	p.namedLogger().With(log.KeySubAccountID, runtime.SubAccountID).With(log.KeyRuntimeID, runtime.RuntimeID).
+		Infof("Not tracking cluster of state: %s, provisioning state: %s, deprovisioning state: %s",
+			runtime.Status.State, provisioning, deprovisioning)
```

Comment:
As per the comment `// Set provisioning and deprovisioning states if available`, is this true, because I see `getOrDefault` then logging, so no set operation was done. Also the function name implies that it checks if a cluster is trackable or not, IMO it should not internally set anything. What do you think?

Reply:
`runtime.Status.Provisioning` and `runtime.Status.Deprovisioning` can be nil. If we try to log them, when they are nil, we get an error. As a work around, the logic `getOrDefault` returns an empty string if there is not available provisioning or deprovisioning status. 

I changed the comment to `Get provisioning and deprovisioning states if available otherwise return empty string` to be more clear.